core/src/env/srml/types.rs

```diff
@@ -34,17 +34,15 @@ pub enum DefaultSrmlTypes {}
 /// For calling into the runtime, a user defined Call type required.
 /// See https://github.com/paritytech/ink-types-node-runtime.
 mod private {
-    use parity_codec::{Encode, Decode, Input};
-
     #[cfg_attr(feature = "std", derive(Debug, Clone, PartialEq, Eq))]
     pub struct Call {
         _unconstructable: ()
     }
-    impl Encode for Call {}
+    impl parity_codec::Encode for Call {}
 
     #[cfg(feature = "std")]
-    impl Decode for Call {
-        fn decode<I: Input>(_value: &mut I) -> Option<Self> {
+    impl parity_codec::Decode for Call {
+        fn decode<I: parity_codec::Input>(_value: &mut I) -> Option<Self> {
```

Comment:
Why do we impl `Decode` for `Call` if we never decode it?

Reply:
It's "used" in the test env so is required as a trait bound on `EnvTypes` https://github.com/paritytech/ink/blob/8c6b85235131d262e3a016842d361d6c009f96a8/core/src/env/test_env.rs#L503.

Suppose we could add the `Decode` bound to a generic type parameter `Call` to just this method - what do you think?

Reply:
Then better put it behind `test-env` feature instead of `std` feature and also provide a rational (comment) why it is there.